src/Storages/StorageMergeTree.cpp

```diff
@@ -1831,9 +1855,16 @@ MutationCommands StorageMergeTree::getFirstAlterMutationCommandsForPart(const Da
     std::lock_guard lock(currently_processing_in_background_mutex);
 
     auto it = current_mutations_by_version.upper_bound(part->info.getDataVersion());
-    if (it == current_mutations_by_version.end())
-        return {};
-    return it->second.commands;
+    while (it != current_mutations_by_version.end())
```

Comment:
Why do we need to change this function at all? `getFirstAlterMutationCommandsForPart` is something too specific and seems like it used only for ALTER METADATA, not for ALTER UPDATE/DELETE. Is it possible to ADD COLUMN IN PARTITION, for example? (seems like it's not). I do not understand why these changes are related to ALTER UPDATE/DELETE IN PARTITION.

Reply:
This function (unexpectedly) returns next (upcoming) mutation for specified part. I would leave it as in branch for consistency at least. Otherwise `getFirstAlterMutationCommandsForPart()` would return irrelevant mutations (for different partitions).